Update k8s 1.17

timwalls opened the pull request:
Hey, hi - thanks for the Helm chart for Mastodon - saved me a hell of a lot of time.

I had to make a few changes to work with the latest Kube API version, and also for the latest docker hub image (it's now tootsuite/mastodon).  If they are useful, feel free to merge.

Thanks again,
Tim.

timwalls committed:
Missing redis password

timwalls committed:
Fixes for environment - SAFETY_ASSURED required to get the latest Mastodon image to run the database install scripts.  LOCAL_HTTPS disable stops the backend instance getting into a horrible SSL redirect loop (we handle SSL at the ingress, so this should be OK.)

timwalls committed:
Fixes for K8s v1.17 API spec.  Also a couple of bugfixes (volume mount name in Sidekiq spec.)

Also unintentionally my changes to tune performance on an underpowered cluster - which I will tidy up in the next commit...

timwalls committed:
Parameterised some performance tuning adjustments, and added reasonable defaults to values.yaml.sample.

timwalls committed:
Oops - removed a little local-only requirement to get install to work (a long helm timeout for my slow cluster...)

timwalls committed:
Fixed Docker image name.

timwalls committed:
Enable serving static content

timwalls committed:
Sidekiq needs to mount the system directory for image processing to work if your static content is delivered by Rust.

timwalls committed:
Updated to a current (v3.2.0) image of Mastodon, and v6 of Redis (Sidekiq won't work with anything older than v4 now.)

eldondev commented:
Hey @timwalls I stumbled across your blog post when I was looking to install mastodon on my own selfhosted kubernetes cluster. Your work saved _me_ a hell of a lot of time. I noticed that there appears to be a nightly cronjob included, but it looks like that is no longer [necessary](https://github.com/tootsuite/mastodon/pull/2495). Would you like a PR to remove this to go on your repo? Did you also run into the cron failing?

timwalls commented:
Hey, hi @eldondev, glad to hear it helped!  And yes, oops, now you make me look at that cronjob for the first time I see that it has been dutifully failing every night with:

` Sept 14 03:00:30 mastodon-daily-cron mastodon[mastodon-daily-cron-1600041600-54hlq] Don't know how to build task 'mastodon:daily' (See the list of available tasks with `rake --tasks`) 
error`

Thanks for the heads-up!  I'll remove the cronjob :).

timwalls committed:
AS @eldon noticed, the daily cronjob is no longer needed- replaced with a job scheduled using Sidekiq (https://github.com/tootsuite/mastodon/pull/2495).  Removing the deployment.

slapula commented:
EDIT: Disregard! I answered my own question. Thanks for working on this!

timwalls committed:
Updating example values.yaml with Mastodon v3.2.1

(Confirmed no other changes needed on my own test system - helm chart deploys this release fine.)
